src/NLopt.jl

```diff
@@ -633,9 +630,10 @@ end
 optimize(o::Opt, x::AbstractVector{<:Real}) =
   optimize!(o, copyto!(Array{Cdouble}(undef,length(x)), x))
 
-############################################################################
-
-include("MPB_wrapper.jl")
-include("MOI_wrapper.jl")
+if !isdefined(Base, :get_extension)
+    include("../ext/MathOptInterfaceExtension.jl")
+    using .MathOptInterfaceExtension
+    const global Optimizer = MathOptInterfaceExtension.Optimizer
```

Comment:
Why is this `global` needed ?

Reply:
Because `Optimizer` is now defined in the extension submodule and needs to be brought up into the top level scope. This is the workaround for https://github.com/JuliaOpt/NLopt.jl/pull/204#issuecomment-1676076939.

Previously, this global was made directly by the struct definition.

Reply:
Yes but I mean you could do `const Optimizer` instead of `const global Optimizer` can't you ?

Reply:
Sure, the `if` block doesn't introduce a new scope, so it's still a global. I've removed the modifier.